### ISSUE TITLE
Difference in prediction results between just trained model and reloaded model

### ISSUE THREAD

**Issue by nanomolar** [deepchem/deepchem#2803]
## ❓ Questions & Help

Hi, I have trained a GCN model successfully, however, the prediction results between just trained model and reloaded model are different. 

Could anyone tell me why? 

Thanks in advance.

```

import deepchem as dc

# data and featurize
data_file = "/home/wemol/deepchem/bace.csv"
featurizer = dc.feat.MolGraphConvFeaturizer(use_edges=True)
loader = dc.data.CSVLoader(['Class'],feature_field="mol",featurizer=featurizer)
dataset = loader.create_dataset(data_file)
print("csv loader",len(dataset))

# split
splitter = dc.splits.RandomSplitter()
train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(seed=123,
  dataset=dataset, frac_train=0.8, frac_valid=0.1, frac_test=0.1
)
print(len(train_dataset),len(valid_dataset),len(test_dataset))

# train
model = dc.models.GCNModel(n_tasks=1, mode='classification', graph_conv_layers=[64,64], dropout=0.5,model_dir="/home/wemol/deepchem/save_model")
model.fit(train_dataset, nb_epoch=50)

# evaluate
metric = dc.metrics.Metric(dc.metrics.roc_auc_score)
print('Training set score:', model.evaluate(train_dataset, [metric]))
print('Test set score:', model.evaluate(test_dataset, [metric]))

# use local model
model2 = dc.models.GCNModel(1, mode="classification", model_dir="/home/wemol/deepchem/save_model")
print('Test set score:', model2.evaluate(test_dataset, [metric]))

```

Output:
csv loader 1513
1210 151 152
Using backend: pytorch
Training set score: {'roc_auc_score': 0.853694557494654}
Test set score: {'roc_auc_score': 0.8189640035118524}
Test set score: {'roc_auc_score': 0.5690956979806847}

**Comment by nanomolar** [comment]
I have test the save() and reload(), both reported error

**Comment by TheRealSalmon** [comment]
It looks like save() is not implemented for GraphConvModel. I will give it a try, I'll update you if I succeed.

**Comment by rbharath** [comment]
In general models autosave when `fit()` is called (the `save()` method is a vestigial remnant of older implementations that we should clean up). Try `restore()` instead of `reload()` as well
